faf-commons-data/src/main/java/com/faforever/commons/replay/header/GameOptions.java

```diff
@@ -0,0 +1,75 @@
+package com.faforever.commons.replay.header;
+
+/**
+ * Populated by the game options field of the table that is passed to `CLobby:LaunchGame`
+ * @param autoTeams
+ * @param teamLock
+ * @param teamSpawn
+ * @param allowObservers
+ * @param cheatsEnabled
+ * @param prebuiltUnits
+ * @param revealedCivilians
+ * @param scoreEnabled
+ * @param unitCap
+ * @param unRated
+ * @param victory
+ */
+public record GameOptions(AutoTeams autoTeams, TeamLock teamLock, TeamSpawn teamSpawn, boolean allowObservers,
+                          boolean cheatsEnabled, boolean prebuiltUnits, boolean revealedCivilians, boolean scoreEnabled,
+                          int unitCap, boolean unRated, Victory victory) {
+
+  public enum AutoTeams {
+    none("None"),
+    manual("Manual"),
+    tvsb("Top versus bottom"),
+    lvsr("Left versus right"),
+    pvsi("Even versus uneven");
+
+    private final String string;
+
+    AutoTeams(String string) {
+      this.string = string;
+    }
+  }
+
+  public enum TeamLock {
+    locked("Locked"), unlocked("Unlocked");
+
+    private final String string;
+
+    TeamLock(String string) {
+      this.string = string;
+    }
+  }
+
+  public enum TeamSpawn {
+    fixed("Fixed"),
+    random("Random"),
+    balanced("Balanced"),
+    balanced_flex("Flexible balanced"),
+    random_reveal("Random and revealed"),
+    balanced_reveal("Balanced and revealed"),
+    balanced_reveal_mirrored("Mirror balanced and revealed"),
+    balanced_flex_reveal("Flexible balanced and revealed");
+
+    private final String string;
+
+    TeamSpawn(String string) {
+      this.string = string;
+    }
+  }
+
+  public enum Victory {
+    demoralization("Assasination"),
+    domination("Supremacy"),
+    eradication("Annihilation"),
+    sandbox("Sandbox");
+
+    private final String string;
+
+    Victory(String string) {
+      this.string = string;
+    }
+  }
```

Comment:
Java enums should be SCREAMING_SNAKE_CASE and AutoTeams can be the full name for the enum

Reply:
Also the strings don't have getters so aren't accessible outside the enum

Reply:
How would I then match the game options with the enums? At the moment they are `demoralization`, `domination`, etc which I think I can immediate map one-to-one to an enum.

Reply:
You would add a findByValue method that takes in a string and returns the enum that matches the string you define as a member of the enum.

Reply:
Noting that the enum names should be normalized

Reply:
Processed in https://github.com/FAForever/faf-java-commons/pull/143/commits/cce2ee3c499805baee7ec85a5813e72703787c48